src/datajudge/constraints/stats.py

```diff
@@ -72,24 +65,26 @@ def c(alpha: float):
         )
 
     @staticmethod
-    def calculate_statistic(engine, table1, table2) -> Any:
+    def calculate_statistic(engine, table1_def: tuple, table2_def: tuple) -> Any:
```

Comment:
too vague and weak annotations

Reply:
What do you think about 
```py
def calculate_statistic(
        engine,
        table1_def: tuple[Selectable | str, str],
        table2_def: tuple[Selectable | str, str],
    ) -> tuple[float, Optional[float], int, int]:
```

Reply:
Better. I guess `Selectable` is a `sqlalchemy` type. 

Remember that we're still supporting some older versions of python, thus the `|` should be `Union[Selectable, str]`

Reply:
> > Better. I guess Selectable is a sqlalchemy type.
> 
> Do you know what else should be used? In practice, everything with a sensible **str** works

No, for now, it looks good.